cockatrice/src/pixmapgenerator.cpp

```diff
@@ -112,32 +112,38 @@ QPixmap CountryPixmapGenerator::generatePixmap(int height, const QString &countr
 
 QMap<QString, QPixmap> CountryPixmapGenerator::pmCache;
 
-QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy)
+QPixmap UserLevelPixmapGenerator::generatePixmap(int height, UserLevelFlags userLevel, bool isBuddy, QString privLevel)
 {
-
-    int key = height * 10000 + (int) userLevel + (int) isBuddy;
+    
+    QString key = QString::number(height * 10000 + (int)userLevel + (int)isBuddy) + privLevel;
```

Comment:
If `userLevel` can be 2 or 1, then this collides because `2 + 0 == 1 + 1`

Why not just `height + ":" + userLevel + ":" + isBuddy` or so?